llama-index-integrations/llms/llama-index-llms-ibm/pyproject.toml

```diff
@@ -0,0 +1,59 @@
+[build-system]
+build-backend = "poetry.core.masonry.api"
+requires = ["poetry-core"]
+
+[tool.codespell]
+check-filenames = true
+check-hidden = true
+# Feel free to un-skip examples, and experimental, you will just need to
+# work through many typos (--write-changes and --interactive will help)
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.llms.ibm"
+
+[tool.llamahub.class_authors]
+WatsonxLLM = "llama-index"
+
+[tool.mypy]
+disallow_untyped_defs = true
+# Remove venv skip when integrated with pre-commit
+exclude = ["_static", "build", "examples", "notebooks", "venv"]
+ignore_missing_imports = true
+python_version = "3.10"
+
+[tool.poetry]
+authors = ["IBM"]
+description = "llama-index llms IBM watsonx.ai integration"
+exclude = ["**/BUILD"]
+license = "MIT"
+name = "llama-index-llms-ibm"
```

Comment:
we already have a Watson integration in `llama-index-integrations/llms/llama-index-llms-watsonx` -- is this different?

Reply:
@logan-markewich  Changes contain both LLM and Embeddings integration. Currently, there exist close LLM integration in llamaindex to the proposed one: WatsonX https://docs.llamaindex.ai/en/stable/examples/llm/watsonx/ . However, since the mentioned integration uses old IBM python client sdk (i.e. ibm-watson-machine-learning), does not contain fully functionality of watsonx.ai service (i.a. deployment inferencing) and is not consistent with the IBM watsonx.ai other open-source integrations (e.g. LangChain) we decided to introduce a new package: `llama-index-llms-ibm`. Making the proposed changes in `llama-index-llms-watsonx` package instead of creating a new integration package may cause a  few breaking changes. Simultaneously, we add deprecation warnings to the current llm WatsonX integration and change documentation accordingly.

Please note that this would be an official IBM watsonx.ai integration. Also, if the PR will be positively reviewed quickly, please DO NOT merge yet - internal request with the official permission is processed.